### ISSUE TITLE
Improve vertical text alignment of OxyPlot.SvgExporter

### ISSUE THREAD

**Issue by Jonarw** [oxyplot/oxyplot#1531]
See discussion in #1518.
Output of SkiaSharp SVG export could give some inspiration.

**Comment by VisualMelon** [comment]
It looks like the Skia svg exporter doesn't touch the `dominant-baseline` alignment, which is the svg parameter that has little browser support. It 'defaults' to baseline, which is a problem, because the Svg exporter has no idea where this is (`MeasureText` only tells it the height). This means the best we can do is to 'guess' where the baseline is (e.g. ~1/4 of the way up) unless we extend some interfaces.

An implementation of this 'just sort of guess where the baseline might be' approach is implemented in 679023676a0247b3e98cbd5da28ba78f48a27d71 and produces results like [these](https://gist.github.com/VisualMelon/946e6b3d5c71e90dbd1b5b28ea562fb0
) when using ImageSharp for text measurement.

![image](https://user-images.githubusercontent.com/5646475/81206281-c84b6380-8fc3-11ea-8311-4534bd32edc8.png)

![image](https://user-images.githubusercontent.com/5646475/81206170-a0f49680-8fc3-11ea-80b9-aa7d039be4d4.png)

That commit also does a few other things:
 - Adds an ImageSharp based Svg exporter
 - Changes everything to use the SVG vertical text workaround by default, since support for dominant-baseline is extremely limited.
 - Improves the multi-line text alignment for SVGs
 - Adds a multi-line aligned text example from #1538 
 - Changes multi-line support in various places to split on the regex "\r?\n" rather than "\r\n"

I'll wait until the nightmare inducing ImageSharp changes have been pulled/rejected/whatever before I add a PR for any of this, and will probably break it up into separate PRs (assuming the descent kludge is acceptable at least for the time being).

While I'm typing, I noticed that the PortableDocumentFont stuff doesn't add gaps between lines (which explains some of the deficiencies in the SvgRenderContext I lazily didn't fix last time round).

**Comment by Jonarw** [comment]
> It 'defaults' to baseline, which is a problem, because the Svg exporter has no idea where this is (`MeasureText` only tells it the height). This means the best we can do is to 'guess' where the baseline is (e.g. ~1/4 of the way up) unless we extend some interfaces.

Could we improve this by overriding `DrawText` in an e.g. ImageSharp-based `SvgRenderContext` implementation? There we have access to more sophisticated text measurements and should be able to properly align the text using the baseline only, right?

**Comment by VisualMelon** [comment]
@Jonarw yeah, good idea. I keep thinking we'd have to modify `IRenderContext` to do this properly, but there's nothing stopping us doing Svg specific work as you suggest.
